contract/models/contract.py

```diff
@@ -217,12 +212,61 @@ def action_contract_send(self):
             'context': ctx,
         }
 
+    @api.model
+    def _get_recurring_create_invoice_domain(self, contract=False):
+        domain = []
+        date_ref = fields.Date.context_today(self)
+        if contract:
+            contract.ensure_one()
+            date_ref = contract.recurring_next_date
+            domain.append(('id', '=', contract.id))
```

Comment:
I don't understand the use of this part (if contract...).

Reply:
@sbidoul Me neither actually, I just kept the existing logic. @sbejaoui ?

Reply:
If create_recurring_invoice is called for one specific contract, date_ref take the contract recurring_next_date and not today

Reply:
Yeah, that relates to my other comment below: https://github.com/sbejaoui/contract/pull/4/files#r242900895

The fact that this logic is hidden here makes the code hard to understand.

Reply:
And actually, it looks like we don't use this _get_contract_to_invoice_domain anymore in create_recurring_invoice.